### PR TITLE
More UI fuxing about

### PR DESCRIPTION
**Pull request by jdillon**
Adds a max-size to the LogEvent store.

Adds stateful handling of dev-panel logging config.

Changes NX.Assert to use NX.Console, avoid potential problems if logging freaks out.

Log state change objects with Ext.encode, until logging framework does this for you.

Add rowexpander to dev-panel logging to show deeper rendering, LogEvent.message is now an array.

Refines dev-panel maximimze to work _better_, but is still busted with the Style tab, and is bitching about some layout failures, but will at least not corrupt the UI after maxinize and minimize cycles (I think).

Changes message notification windows to stick on hover (so if you mouse over a message it will not vanish).

Removes messages panels (dev and unused app panels) and the store/model which were not used and would lead to memory issues in the browser.

Adds NX.app.Controller to define core extensions (so we can document them) and configure logging.

### CODE REVIEW

**Comment by adreghiciu** [comment]
+1

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/dev/Logging.js [hunk 1]
```diff
@@ -42,34 +38,64 @@ Ext.define('NX.controller.dev.Logging', {
...
-          select: function(combo) {
-            me.getController('Logging').setThreshold(combo.getValue());
+
+        'nx-dev-logging numberfield[itemId=bufferSize]': {
+          afterrender: function (field) {
+            field.setValue(sink('store').maxSize);
           },
-          afterrender: function(combo) {
-            combo.select(me.getController('Logging').getThreshold());
+          change: function (field) {
```
Comment (kellyrob99) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/dev/Logging.js]: should be able to defer reacting to changes here if you demand an ENTER to set the value.
Switch to http://docs.sencha.com/extjs/4.2.3/#!/api/Ext.dom.Element-event-keypress and wrap the mutation of the sink with  if(event.getKey() == event.ENTER)

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/view/dev/Styles.js [hunk 2]
```diff
@@ -75,15 +78,27 @@ Ext.define('NX.view.dev.Styles', {
...
       ];
 
-      Ext.each(sections, function (section) {
-        var component = Ext.create('NX.view.dev.styles.' + section);
-        me.add(component);
+      me.logDebug('Creating style guide');
+
+      // TODO: See if suspending layouts here actually helps anything?
+      //Ext.AbstractComponent.suspendLayouts();
+      //try {
```
Comment (jdillon) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/view/dev/Styles.js]: FTR I'm leaving this in as reminder on how to properly suppsend/resume layouts... we may need to apply this to other parts that do this.  Unsure however if this does anything at all for this specific bit of code.

**Comment by kellyrob99** [comment]
+1

**Comment by mcculls** [comment]
looks reasonable +1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**